### PR TITLE
Add distributed wrangler

### PR DESCRIPTION
**Pull request by gaohao95**
Needs:
- [x] Depends on https://github.com/inducer/boxtree/pull/55.
- [x] Point `requirements.txt` back to boxtree main

### CODE REVIEW

File: sumpy/fmm.py [hunk 1]
```diff
@@ -1083,6 +1083,95 @@ def finalize_potentials(self, potentials, template_ary):
 # }}}
 
 
+# {{{ distributed expansion wrangler
+
+try:
+    from boxtree.distributed.calculation import DistributedExpansionWrangler
+    is_distributed_wrangler_available = True
+except ModuleNotFoundError:
+    is_distributed_wrangler_available = False
```
Comment (inducer) [inline sumpy/fmm.py:1092]: Move to separate file instead.

**Comment by inducer** [comment]
LGTM, thanks!

File: .test-conda-env-py3.yml [hunk 2]
```diff
@@ -16,3 +16,4 @@ dependencies:
 - pyfmmlib
 - pyrsistent
 - pyvkfft
+- mpi4py
```
Comment (isuruf) [inline .test-conda-env-py3.yml:19]: To be explicit about the test dependency
```suggestion
- mpi4py
- platformdirs
```

Comment (inducer) [reply .test-conda-env-py3.yml:19]: Good catch. Sorry this missed the merge! Is there something like `test_requires` in `setup.py`, too?

Comment (isuruf) [reply .test-conda-env-py3.yml:19]: There was `tests_require`, but that was deprecated and removed I think. Don't think there's any other option setup.py

Comment (inducer) [reply .test-conda-env-py3.yml:19]: #140

**Pull request merged**